streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -589,6 +589,10 @@ private void subscribeConsumer() {
         }
     }
 
+    public void resizeCache(final long size) {
+        taskManager.resizeCache(size);
```

Comment:
I am not sure if I should just expose cache or pass it along. It is about 4 levels deep

Reply:
I am in favour of keeping a reference to the thread cache in the `StreamThread` and do the re-sizing here. I think it makes the code a bit easier to follow. 
You will need synchronization, because the thread that will add the new stream thread will also resize the thread caches.

Reply:
Are the thread caches not independent of each other? Also I was not planning on having the new thread resize the cache but the calling thread do so

Reply:
> I was not planning on having the new thread resize the cache but the calling thread do so

That is what I am saying "the thread that will add the new stream thread" is the calling thread. The new stream thread cannot resize the caches of the other stream threads because it is not aware of the other stream threads. Still we need synchronization because the calling thread will access and modify the thread caches of all stream threads and all stream threads will access and modify their own thread cache during normal processing.

Reply:
I'm not sure I follow. If we make it so that each thread is responsible for resizing their own independent cache then why would the method need to synchronized as each call should not affect the others. And if one thread does all the work there should only be one call. Unless you are thinking about multiple threads adding threads at the same time?

Maybe I don't understand how the cache is set up well enough?

Reply:
Talked about off line. The named cache is already synchronized. Also made max cache size volatile